apps/web/app/post/[id]/page.tsx

```diff
@@ -357,26 +363,49 @@ const Post = async ({ params }: PostProps) => {
                 (m) => m.snowflakeId === post.answerId,
               )}
             >
-              {group.messages.map((message, i) => (
-                <Message
-                  key={message.id.toString()}
-                  snowflakeId={message.snowflakeId}
-                  createdAt={message.createdAt}
-                  content={message.content}
-                  isFirstRow={i === 0}
-                  author={{
-                    username: message.authorUsername,
-                    avatarUrl: message.authorAvatarUrl,
-                    isPublic: message.userIsPublic,
-                    isOP: postMessage
-                      ? message.authorId === postMessage.authorId
-                      : false,
-                    isModerator: message.userIsModerator,
-                    userID: message.userID,
-                  }}
-                  attachments={message.attachments}
-                />
-              ))}
+              {group.messages.map((message, i) => {
+                const hasReply = message.replyToMessageId !== null
+                const replyData = hasReply
+                  ? allMessages.find(
+                      (m) => m.snowflakeId === message.replyToMessageId,
+                    )
+                  : undefined
```

Comment:
This will simplify the ternary on the JSX, also avoid using "data" as a variable name because most of the time it is too generic to be meaningful
```suggestion
                const replyMessage = hasReply
                  ? allMessages.find(
                      (m) => m.snowflakeId === message.replyToMessageId,
                    )
                  : "deleted"
```